File: webui/src/main/scala/com/taskadapter/webui/EventTracker.scala
package com.taskadapter.webui

import com.taskadapter.web.event.{ApplicationActionEvent, ApplicationActionEventWithValue, EventBusImpl}

object EventTracker {
  def trackEvent(category: EventCategory, action: String, label: String): Unit = {
    EventBusImpl.post(ApplicationActionEvent(category, action, label))
  }

  def trackEvent(category: EventCategory, action: String, label: String, value: Int): Unit = {
    EventBusImpl.post(ApplicationActionEventWithValue(category, action, label, value))
  }
}

File: webui/src/main/java/com/taskadapter/web/event/EventTracker.java
package com.taskadapter.web.event;

public class EventTracker {
    public static void trackEvent(EventCategory category, String action, String label) {
        EventBusImpl.post(new ApplicationActionEvent(category, action, label));
    }

    public static void trackEvent(EventCategory category, String action, String label, int value) {
        EventBusImpl.post(new ApplicationActionEventWithValue(category, action, label, value));
    }
}
